include/cuda/std/memory_resource

```diff
@@ -239,17 +239,17 @@ private:
   /// Default synchronous implementation of `memory_resource::do_allocate`
   void *do_allocate(std::size_t __bytes, std::size_t __alignment) override {
     auto const __default_stream = stream_view{};
+    __default_stream.synchronize();
     auto __p = do_allocate_async(__bytes, __alignment, default_stream);
```

Comment:
Seems like to implement synchronous allocation in terms of asynchronous allocation, you still need to sync after the `do_allocate_async`, right? Because the caller of a synchronous method is expecting the memory allocated to be immediately accessible from any stream as well as from the host, but `do_allocate_async(... default_stream)` only guarantees that the memory is accessible immediately on `default_stream`.

Reply:
That's true:
Thread A: sychronize
Thread B: free on default stream
Thread A: allocate on default stream
In this scenario, there might be some outstanding work on the stream.

Now I wonder if there's a need for synchronization _before_ allocation.